File: flowquery-py/src/parsing/operations/delete_node.py
"""Represents a DELETE operation for deleting virtual nodes."""

from typing import Any, Dict, List

from ...graph.database import Database
from ...graph.node import Node
from .operation import Operation


class DeleteNode(Operation):
    """Represents a DELETE operation for deleting virtual nodes."""

    def __init__(self, node: Node) -> None:
        super().__init__()
        self._node = node

    @property
    def node(self) -> Node:
        return self._node

    async def run(self) -> None:
        if self._node is None:
            raise ValueError("Node is null")
        db = Database.get_instance()
        db.remove_node(self._node)

    @property
    def results(self) -> List[Dict[str, Any]]:
        return []

File: src/parsing/operations/delete_node.ts
import Database from "../../graph/database";
import Node from "../../graph/node";
import Operation from "./operation";

class DeleteNode extends Operation {
    private _node: Node | null = null;
    constructor(node: Node) {
        super();
        this._node = node;
    }
    public get node(): Node | null {
        return this._node;
    }
    public run(): Promise<void> {
        return new Promise(async (resolve, reject) => {
            try {
                if (this._node === null) {
                    throw new Error("Node is null");
                }
                const db: Database = Database.getInstance();
                db.removeNode(this._node);
                resolve();
            } catch (error) {
                reject(error);
            }
        });
    }
    public get results(): Record<string, any>[] {
        return [];
    }
}

export default DeleteNode;
